server/methods/core/orders.js

```diff
@@ -139,6 +139,18 @@ Meteor.methods({
       throw new Meteor.Error(403, "Access Denied");
     }
 
+    // Server-side check to make sure discount is not greater than orderTotal.
+    let orderTotal =
+      order.billing[0].invoice.subtotal
+      + order.billing[0].invoice.shipping
+      + order.billing[0].invoice.taxes;
+
+    if (discount > orderTotal) {
+      const err = "Discount is greater than the cart total";
+      Logger.error(err);
+      throw new Meteor.Error("orders/approvePayment.discount-amount", err);
```

Comment:
just nitpicking, but I'd rather see `error` than `err` or `e`. in this case, it doesn't bother me, but it does bother me on try/catch statements.